saleor/checkout/complete_checkout.py

```diff
@@ -354,7 +354,7 @@ def _create_order_line_discounts(
 
 def _get_sale_id(line_discounts: list[OrderLineDiscount]):
     for discount in line_discounts:
-        if discount.type == DiscountType.PROMOTION:
+        if discount.type == DiscountType.CATALOGUE_PROMOTION:
```

Comment:
Here on v3.19 we need to handle both situations when discount type is `PROMOTION` and `CATALOGUE_PROMOTION`, as the type is updated in the background so the value might be `PROMOTION` or `CATALOGUE_PROMOTION`. Only on the main branch we can be sure that instances have only new `CATALOGUE_PROMOTION` type

```suggestion
        if discount.type in [DiscountType.CATALOGUE_PROMOTION, DiscountType.PROMOTION]:
```